_drafts/nrf-connect-with-github-actions.md

```diff
@@ -0,0 +1,859 @@
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/ccache
+          key: ccache-v2-${{ runner.os }}-${{ hashFiles('zephyr-workspace/nrfconnect-ci-app/west.yml') }}
+          restore-keys: |
+            ccache-v2-${{ runner.os }}-
+
+      - name: 🔨 Build Project
+        run: |
+          cd zephyr-workspace
+          ccache -z
+          west build \
+              --board nrf7002dk_nrf5340_cpuapp \
+              --pristine=always nrfconnect-ci-app \
+              -- \
+              -DCONFIG_MEMFAULT_NCS_PROJECT_KEY=\"1234\"
+          ccache -sv
+```
+{% endraw %}
+
+This runs in `1m44s` when cached, which is pretty good! The custom docker image
+is about 3.4GB uncompressed, which is a lot smaller than the Zephyr general
+purpose image.
+
+Overall advantages to our custom image:
+
+- we can control exactly what software is installed
+- smaller image size (more precisely fit to our needs)
+- we can include a Zephyr workspace cache in the image, to speed up the
+  `west update` step
+
+The downside is now we're on the hook for maintaining our Docker image- the
+convenience of the GitHub container registry does make this a little less
+painful, though.
+
+### Fine Tuning Docker Images
+
+I'll mention a few tools I use to tighten up Docker images:
+
+- [dive](https://github.com/wagoodman/dive) - a sophisticated tool that is used
+  to analyze each layer in a Docker image, and identify opportunities to reduce
+  unused items
+- [dua](https://github.com/Byron/dua-cli) - this is actually not
+  Docker-specific, but it's a great tool for quickly drilling through the Docker
+  image file system, checking for large files that can be removed
+
+Here's a way to use `dua` to inspect the file system of our image:
+
+```bash
+❯ docker run --rm -i -t \
+  ghcr.io/noahp/memfault-nrfconnect-ci-app:2024-01-17 \
+    bash -c 'cd /tmp && wget -q -c https://github.com/Byron/dua-cli/releases/download/v2.17.5/dua-v2.17.5-x86_64-unknown-linux-musl.tar.gz -O - | tar -xz && ./dua*/dua i /'
+```
+
+Which yields an interactive view of the file system:
+
+![](/img/ncs-github-actions/dua-custom-image.png)
+
+## Outro
+
+That's it for this article! We've covered a lot of specifics around setting up
+and tuning GitHub actions for building nRF-Connect SDK projects. I know I
+learned a good bit about optimizing nRF-Connect SDK builds for CI, and I have a
+much better understanding of how the Zephyr SDK components work together when
+building a project.
+
+I'll leave this XKCD comic here, which I think nicely summarizes what we worked
+through in this article 😄:
+
+[![](https://imgs.xkcd.com/comics/containers.png)](https://xkcd.com/1988/)
+
+<!-- Interrupt Keep START -->
+{% include newsletter.html %}
+
+{% include submit-pr.html %}
+<!-- Interrupt Keep END -->
+
+{:.no_toc}
+
+## References
+
+<!-- prettier-ignore-start -->
+- [Nordic nRF-Connect SDK](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/2.5.1/nrf/index.html)
```

Comment:
I'd also drop a link to the repo you were experimenting with! In case people want to poke around the logs yourself (and you don't mind that)

Reply:
oh :facepalm: of course! will add that, thank you!